src/workers/personalizedDigestEmail.ts

```diff
@@ -0,0 +1,165 @@
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+}): Date => {
+  const nextPreferredDay = nextDay(new Date(), personalizedDigest.preferredDay);
+  nextPreferredDay.setHours(personalizedDigest.preferredHour, 0, 0, 0);
+  const sendDateInPreferredTimezone = zonedTimeToUtc(
+    nextPreferredDay,
+    personalizedDigest.preferredTimezone,
+  );
+
+  return sendDateInPreferredTimezone;
+};
+
+const getPreviousSendDate = ({
+  personalizedDigest,
+}: {
+  personalizedDigest: UserPersonalizedDigest;
+}): Date => {
+  const nextPreferredDay = previousDay(
+    new Date(),
+    personalizedDigest.preferredDay,
+  );
+  nextPreferredDay.setHours(personalizedDigest.preferredHour, 0, 0, 0);
+  const sendDateInPreferredTimezone = zonedTimeToUtc(
+    nextPreferredDay,
+    personalizedDigest.preferredTimezone,
+  );
+
+  return sendDateInPreferredTimezone;
+};
+
+const getPostsTemplateData = async ({ posts }: { posts: ArticlePost[] }) => {
+  const templateData = await Promise.all(
+    posts.map(async (post) => {
+      const source = await post.source;
+
+      const postUrl = new URL(getDiscussionLink(post.id));
+      postUrl.searchParams.append('utm_source', 'notification');
+      postUrl.searchParams.append('utm_medium', 'email');
+      postUrl.searchParams.append('utm_campaign', 'digest');
+
+      return {
+        post_title: post.title,
+        post_image: post.image || pickImageUrl(post),
+        post_link: postUrl.toString(),
+        source_name: source.name,
+        source_image: source.image,
+      };
+    }),
+  );
+
+  return templateData;
+};
+
+const worker: Worker = {
+  subscription: 'api.personalized-digest-email',
+  handler: async (message, con, logger) => {
+    if (process.env.NODE_ENV === 'development') {
+      return;
+    }
+
+    const data = messageToJson<Data>(message);
+    const { personalizedDigest } = data;
+
+    const user = await con.getRepository(User).findOneBy({
+      id: personalizedDigest.userId,
+    });
+
+    if (!user || !user.infoConfirmed) {
+      return;
+    }
+
+    const currentDate = new Date();
+    const emailSendDate = getEmailSendDate({ personalizedDigest });
+    const previousSendDate = getPreviousSendDate({ personalizedDigest });
+
+    const feedConfig = await feedToFilters(
+      con,
+      personalizedDigest.userId,
+      personalizedDigest.userId,
+    );
+
+    const feedResponse = await personalizedDigestFeedClient.fetchFeed(
+      { log: logger },
+      personalizedDigest.userId,
+      {
+        user_id: personalizedDigest.userId,
+        total_posts: personalizedDigestPostsCount,
+        date_from: format(previousSendDate, personalizedDigestDateFormat),
+        date_to: format(currentDate, personalizedDigestDateFormat),
+        allowed_tags: feedConfig.includeTags,
+        blocked_tags: feedConfig.blockedTags,
+        blocked_sources: feedConfig.excludeSources,
+      },
+    );
+
+    const posts = await con.getRepository(ArticlePost).find({
```

Comment:
We need to order it by the same order we receive from the feed

Reply:
Good point.